rust/processor/src/gap_detectors/mod.rs

```diff
@@ -28,18 +28,26 @@ pub enum GapDetector {
 
 #[enum_dispatch(GapDetectorTrait)]
 pub enum GapDetectorResult {
-    DefaultGapDetectorResult,
-    ParquetFileGapDetectorResult,
+    DefaultGapDetectorResult(DefaultGapDetectorResult),
```

Comment:
I can't comment on it but on line 23, is it meaningful to have `#[enum_dispatch(GapDetectorTrait)]` on `GapDetector`? It's an enum without variants, what is it dispatching to? I tried removing it locally and the code was still fine.

Reply:
Ah actually, seems like we should expand the use of this enum. See my comment on line 329 in rust/processor/src/worker.rs.